refactor(linter): simplify error handling

camchenry opened the pull request:
adds some helper functions to simply the error handling for the oxlint CLI. not perfect by any means, but at least a little simpler to read.

camchenry commented:
* **#10404** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10404?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10404?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10404).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Sysix commented on apps/oxlint/src/lint.rs:
```diff
@@ -561,6 +526,26 @@ impl LintRunner {
+    }
+}
+
+fn render_report(handler: &GraphicalReportHandler, diagnostic: &OxcDiagnostic) -> String {
```
Thanks for simplifying it. This was always something that bothered me. ❤️ 

This is a bit deeper offtopic: In my perfect world without problems:
`GraphicalReportHandler` should be used only in `output_formatter::default`. 
The JSON output wants report an error in JSON. 
We should still provide a default implementation for other output formats.

camchenry replied:
yeah, I agree, the concerns here are a little bit mixed. I thought about trying to refactor this with the linter runtime work as well, but decided to leave it.

Sysix commented:
### Merge activity

* **Apr 18, 12:13 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camchenry committed:
refactor(linter): simplify error handling (#10404)

adds some helper functions to simply the error handling for the oxlint CLI. not perfect by any means, but at least a little simpler to read.
